lxd/cluster/heartbeat.go

```diff
@@ -19,12 +19,13 @@ import (
 
 // APIHeartbeatMember contains specific cluster node info.
 type APIHeartbeatMember struct {
-	ID            int64
-	Address       string
-	Raft          bool
-	LastHeartbeat time.Time
-	Online        bool // Calculated from offline threshold and LastHeatbeat time.
-	updated       bool // Has node been updated during this heartbeat run. Not sent to nodes.
+	ID            int64     // ID field value in nodes table.
+	Address       string    // Host and Port of node.
+	RaftID        int64     // ID field value in raft_nodes table, zero if non-raft node.
```

Comment:
Is wanting to have the exact same IDs across all raft_nodes tables the only reason we are sending the RaftID field?

I assume this is being done because the old heartbeat mechanism did indeed preserve the database ID, however I **believe** that it's not strictly required, and things should work even with random IDs assigned by the database. It'd be good to double check what we use the raft_nodes.id field for, probably nothing.

Reply:
@freeekanayaka that is correct, we need to populate the raft_nodes table consistently across all nodes, yet still send the node ID from the all nodes table for all other nodes to consume, and most importantly, we need to make sure that the unified list isn't keyed on ID but instead address, such that there are no duplicates created when a node has a different ID in each table.